zerver/tests/test_subs.py

```diff
@@ -719,6 +810,53 @@ def test_bad_delete_parameters(self):
         )
         self.assert_json_error(result, "delete[0] is not a string")
 
+    def test_add_or_delete_not_specified(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        self.login(email)
+
+        result = self.client_patch(
+            "/api/v1/users/me/subscriptions",
+            {},
+            **self.api_auth(email)
+        )
+        self.assert_json_error(result,
+                               'Nothing to do. Specify at least one of "add" or "delete".')
+
+    def test_invalid_stream_name(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        self.login(email)
+
+        invalid_stream_name = ""
```

Comment:
I think we removed the alphanumeric check to properly support unicode stream names (relevant for teams where English isn't the primary language).  Other than checking for too-long names, I'm not sure what else we can reasonably exclude (from the perspective that someone might want that to be their stream name).

We do strip trailing whitespace from stream names; I think we should strip leading whitespace as well if we don't already.